plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1087,29 +1087,133 @@ public void testCaseInsensitivePredicate()
+                .isFullyPushedDown();
+        // column reference with AND
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE regionkey = 0 AND nationkey + regionkey = 14"))
+                .matches("VALUES (BIGINT '14', CAST('KENYA' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE regionkey = 0 AND nationkey - regionkey = 16"))
+                .matches("VALUES (BIGINT '16', CAST('MOZAMBIQUE' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
 
-        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2", errorMessageForDivisionByZeroPushdown());
+        // non-integer types shouldn't get pushed down
+        assertThat(query("SELECT custkey, name, acctbal FROM customer WHERE acctbal + custkey = 524.2"))
+                .isNotFullyPushedDown(FilterNode.class);
+        assertThat(query("SELECT custkey, name, acctbal FROM customer WHERE acctbal - custkey = 119.65"))
+                .isNotFullyPushedDown(FilterNode.class);
 
+        // multiplication/division/modulo by zero
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey * 0 != 0"))
+                .returnsEmptyResult()
+                .isFullyPushedDown();
+        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey / 0 = 0", errorMessageForDivisionByZeroPushdown());
+        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey % 0 = 0", errorMessageForDivisionByZeroPushdown());
         // Expression that evaluates to 0 for some rows on RHS of modulus
+        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) / (regionkey - 1) = 2", errorMessageForDivisionByZeroPushdown());
         assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % (regionkey - 1) = 2", errorMessageForDivisionByZeroPushdown());
 
-        // TODO add coverage for other arithmetic pushdowns https://github.com/trinodb/trino/issues/14808
+        // multiplicative/divisive identity
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey * 1 = nationkey"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey / 1 = nationkey"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        // multiplicative/divisive inverse
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND nationkey / nationkey = 1"))
+                .matches("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0")
```

Comment:
.matches is redundant. 
why do you have it for some methods?

note that .isFullyPushedDown(); already does a correctness check too

Reply:
Also some of them were pre-existing - https://github.com/trinodb/trino/commit/b4331b80c74b0a2b7510e9f0dece836b6e57217c#diff-01f7e5fb80c1ce75cf0ea5d3f1933156eb7984967b99b92090bf931e93569696R713 😆

Reply:
I changed code to not introduce new usages - but I've left the old ones for now - have separate PR to clean them up.